core/src/main/java/org/jdbi/v3/core/statement/Binding.java

```diff
@@ -77,10 +84,16 @@ public void addNamedArgumentFinder(NamedArgumentFinder args) {
     }
 
     /**
-     * Look up an argument by position
+     * @return the set of known binding names
+     */
+    public Collection<String> getNames() {
+        return Collections.unmodifiableSet(names);
```

Comment:
This could be calculated on invocation instead of unconditionally allocating a HashSet for it for every statement.